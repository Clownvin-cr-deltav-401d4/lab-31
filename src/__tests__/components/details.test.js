import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../../components/details';

describe('<Details />', () => {
  it('can render initially in section#details', () => {
    let app = shallow(<Details item={{text: 'Test', difficulty: 3, assignedTo: 'Me', date: 'Today'}} />);
    expect(app.find('section#details').exists()).toBe(true);
    expect(app.find('section#details p.text').exists()).toBe(true);
    expect(app.find('section#details p.difficulty').exists()).toBe(true);
    expect(app.find('section#details p.assigned-to').exists()).toBe(true);
    expect(app.find('section#details p.date').exists()).toBe(true);
    expect(app.find('section#details div header button').exists()).toBe(true);
  });

  it('will not render anything if item not passed in params', () => {
    let app = shallow(<Details />);
    expect(app.find('section#details').exists()).toBe(false);
  });

  it('can close modal by clicking close button', () => {
    // Arrange
    let clicked = false;
    function func() {
      clicked = true;
    }

    let app = mount(<Details item={{text: 'Test', difficulty: 3, assignedTo: 'Me', date: 'Today'}} closeDetails={func} />);
    let button = app.find('section#details div header button');
    
    button.simulate('click', {preventDefault: () => {}});

    //Assert
    expect(clicked).toBeTruthy();
  });

  it('renders consistently', () => {
    const tree = renderer.create(<Details item={{text: 'Test', difficulty: 3, assignedTo: 'Me', date: 'Today'}} />).toJSON();
    // console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});
