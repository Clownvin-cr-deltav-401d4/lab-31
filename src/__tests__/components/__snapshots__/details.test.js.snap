// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders consistently 1`] = `
<section
  id="details"
>
  <div>
    <header>
      <button>
        x
      </button>
    </header>
    <p
      className="text"
    >
      Test
    </p>
    <p
      className="difficulty d3"
    >
      Difficulty: 
      <span>
        3
      </span>
    </p>
    <p
      className="assigned-to"
    >
      Assigned To: 
      <span>
        Me
      </span>
    </p>
    <p
      className="date"
    >
      Date: 
      <span>
        Today
      </span>
    </p>
  </div>
</section>
`;
